src/typ/tests.rs

```diff
@@ -226,6 +226,18 @@ fn infer_test() {
         "let tuple(tuple(_, _) as x, _) = tuple(tuple(0, 1.0), []) x",
         "tuple(Int, Float)"
     );
+    assert_infer!("let x: String = \"\" x", "String");
+    assert_infer!("let x: tuple(x, x) = tuple(5, 5) x", "tuple(Int, Int)",);
+    assert_infer!("let x: tuple(x, y) = tuple(5, 5.0) x", "tuple(Int, Float)",);
+    assert_infer!("let [1, 2, ..x]: List(Int) = [1,2,3] x", "List(Int)",);
+    assert_infer!(
+        "let tuple(5, [..x]): tuple(x, List(x)) = tuple(5, [1,2,3]) x",
+        "List(Int)",
+    );
+    assert_infer!(
+        "let tuple(5.0, [..x]): tuple(y, List(x)) = tuple(5.0, [1,2,3]) x",
```

Comment:
Huh, what Erlang does [..x] generate? I'm not sure that should be valid.

Reply:
Good question, I'll check once I'm on my computer in an hour or so.

Reply:
It looks like it assigns `X` to the entire list.

This:

```
let [..x] = [1,2,3]
```

Generates:

```
X = [1, 2, 3]
```

It's a little bit weird, but I suppose it kind of makes sense, if `..x` means "the rest of the list that `..x` is contained within"?

Reply:
I think maybe we should remove this syntax